src/components/MoneyRequestReportView/MoneyRequestReportTransactionList.tsx

```diff
@@ -216,6 +217,8 @@ function MoneyRequestReportTransactionList({
                 }
             }
 
+            setOptimisticTransactionThread(reportIDToNavigate, report?.reportID, iouAction?.reportActionID, report?.policyID);
```

Comment:
What if the `reportIDToNavigate` was not defined and we had to create new transaction thread - how does it work in that case (also can you please update the code above to use `reportIDToNavigate` instead of `iouAction?.childReportID`

Reply:
Changed the logic to create new transaction thread when there isn't already one, and set optimistic data if it exists